web/src/app/components/cdf/cdf.utils.ts

```diff
@@ -0,0 +1,294 @@
+ * the interface SVGSVGElement. Datum, PElement, and PDatum are unused and thus,
+ * assigned to undefined or null.
+ * The d3G type is similar to the svg element but SVGGElement refers to a g
+ * element within a svg.
+ */
+export type d3SVG = d3.Selection<SVGSVGElement, undefined, null, undefined>;
+type d3G = d3.Selection<SVGGElement, undefined, null, undefined>;
+
+const NANO_TO_MINUTES = (10 ** 9) * 60;
+const COMPLETED_STATE_TAG = 5;
+
+export const COMPLETED_BLUE = '#00bfa5';
+export const STROKE_COLOR = '#167364';
+
+/**
+ * Calculates the duration between the completed stage and the first non-empty
+ * stage. Assigns the probability as the rank of the duration value over the
+ * total number of points. The duration and probability are defined in an
+ * interface and all points stored as an array of CdfData interfaces.
+ *
+ * @param pushInfos Array of pushes for a single push def
+ * @return Array of Items sorted by increasing duration
+ */
+export function populateData(pushInfos: step189_2020.IPushInfo[]): Item[] {
+  const durations: number[] = [];
+  pushInfos.forEach(pushInfo => {
+    if (!pushInfo) { return; }
+    const states = pushInfo.stateInfo;
+    if (!states) { return; }
+    const pushEndTime = states[states.length - 1].startTimeNsec;
+    if (!pushEndTime) { return; }
+    const finalState = states[states.length - 1].state;
+    if (!finalState) { return; }
+
+    if (finalState === COMPLETED_STATE_TAG) {
+      // Find the start time of the first non-empty stage.
+      let firstStateStart: number | Long = -1;
+      for (const state of states) {
+        if (state.stage && state.startTimeNsec) {
+          firstStateStart = state.startTimeNsec;
+          break;
+        }
+      }
+      if (firstStateStart !== -1) {
+        const duration = (+pushEndTime - +firstStateStart) / NANO_TO_MINUTES;
+        durations.push(duration);
+      }
+    }
+  });
+
+  const sortedArray: number[] = durations.sort((n1, n2) => n1 - n2);
+
+  const data: Item[] = [];
+  const durationLength = sortedArray.length;
+  for (let i = 0; i < durationLength; i++) {
+    const duration = sortedArray[i];
+    const probability = (i + 1) / durationLength;
+    data.push({
+      duration,
+      probability
+    } as Item);
+  }
+  return data;
+}
+
+/**
+ * Finds the probability of a duration taking less time than the given duration
+ * parameter. Because the curve is a step function and the data is sorted by
+ * increasing duration, the resulting probability is the probability of the
+ * Item with a duration directly less than the given duration parameter.
+ *
+ * @param data Array of Items sorted by increasing duration
+ * @param duration A duration value between the min and max of all durations
+ * @return A probability value between 0 and 1 (exclusive)
+ */
+export function getProbabilityForDuration(
+                  data: Item[],
+                  duration: number): number {
```

Comment:
These indentation doesn't seem to follow the guidelines from [1]. Same in the other similar places.

[1] https://google.github.io/styleguide/jsguide.html#formatting-function-arguments